azurelinuxagent/common/osutil/default.py

```diff
@@ -331,6 +331,13 @@ def is_sys_user(self, username):
             return True
         else:
             return False
+    
+    def getusers(self):
+        """
+        Gets all users that can log into the machine
+        """
+        users = [i.split(':') for i in open('/etc/shadow').readlines()]
```

Comment:
You have implemented half a pattern here - if this is distro-agnostic, and you are confident in that, this does not need to be behind osutil. If not, then the base implementation should be a no-op with the distro-specific implementation in the appropriate class.

Reply:
Do not read /etc/shadow. Python has a module for this (pwd) which is standard on all Unix, Linux, and Unix-like systems; use that. Once you do so, your code is absolutely distro-agnostic, so it doesn't need to be in osutil.

Reply:
Removed, thank you for the pointer!